static/app/components/modals/discoverAddToDashboardModal.tsx

```diff
@@ -0,0 +1,535 @@
+import * as React from 'react';
+import {browserHistory} from 'react-router';
+import {css} from '@emotion/react';
+import styled from '@emotion/styled';
+import cloneDeep from 'lodash/cloneDeep';
+import isEqual from 'lodash/isEqual';
+import pick from 'lodash/pick';
+import set from 'lodash/set';
+
+import {validateWidget} from 'app/actionCreators/dashboards';
+import {addErrorMessage} from 'app/actionCreators/indicator';
+import {ModalRenderProps} from 'app/actionCreators/modal';
+import {Client} from 'app/api';
+import Button from 'app/components/button';
+import ButtonBar from 'app/components/buttonBar';
+import WidgetQueriesForm from 'app/components/dashboards/widgetQueriesForm';
+import SelectControl from 'app/components/forms/selectControl';
+import {PanelAlert} from 'app/components/panels';
+import {t} from 'app/locale';
+import space from 'app/styles/space';
+import {GlobalSelection, Organization, SelectValue, TagCollection} from 'app/types';
+import {
+  aggregateOutputType,
+  isAggregateField,
+  isLegalYAxisType,
+} from 'app/utils/discover/fields';
+import Measurements from 'app/utils/measurements/measurements';
+import withApi from 'app/utils/withApi';
+import withGlobalSelection from 'app/utils/withGlobalSelection';
+import withTags from 'app/utils/withTags';
+import {DISPLAY_TYPE_CHOICES} from 'app/views/dashboardsV2/data';
+import {
+  DashboardListItem,
+  DisplayType,
+  Widget,
+  WidgetQuery,
+} from 'app/views/dashboardsV2/types';
+import WidgetCard from 'app/views/dashboardsV2/widgetCard';
+import {generateFieldOptions} from 'app/views/eventsV2/utils';
+import Input from 'app/views/settings/components/forms/controls/input';
+import Field from 'app/views/settings/components/forms/field';
+
+export type DiscoverAddToDashboardModalOptions = {
+  organization: Organization;
+  defaultQuery?: string;
+};
+
+type Props = ModalRenderProps &
+  DiscoverAddToDashboardModalOptions & {
+    api: Client;
+    organization: Organization;
+    selection: GlobalSelection;
+    tags: TagCollection;
+  };
+
+type ValidationError = {
+  [key: string]: string[] | ValidationError[] | ValidationError;
+};
+
+type FlatValidationError = {
+  [key: string]: string | FlatValidationError[] | FlatValidationError;
+};
+
+type State = {
+  title: string;
+  displayType: Widget['displayType'];
+  interval: Widget['interval'];
+  queries: Widget['queries'];
+  loading: boolean;
+  errors?: Record<string, any>;
+  dashboards: SelectValue<string>[];
+  selectedDashboard?: SelectValue<string>;
+};
+
+const newQuery = {
+  name: '',
+  fields: ['count()'],
+  conditions: '',
+  orderby: '',
+};
+
+function mapErrors(
```

Comment:
can we just export this [function](https://github.com/getsentry/sentry/blob/master/static/app/components/modals/addDashboardWidgetModal.tsx#L83) instead? maybe move it to a utils file too if it's being used in multiple places.